out/notebooks/spl2/installer.ts

```diff
@@ -0,0 +1,506 @@
+import axios from 'axios';
+import * as child_process from 'child_process';
+import { XMLParser} from 'fast-xml-parser';
+import * as fs from 'fs';
+import * as path from 'path';
+import { pipeline } from 'stream';
+import * as tar from 'tar-fs';
+import * as unzipper from 'unzipper';
+import * as util from 'util';
+import { env, ExtensionContext, StatusBarItem, Uri, window, workspace } from 'vscode';
+import * as zlib from 'zlib';
+
+// Keys used to store/retrieve state related to this extension
+export const configKeyAcceptedTerms = 'splunk.spl2.acceptedTerms';
+export const configKeyJavaPath = 'splunk.spl2.javaPath';
+export const configKeyLspVersion = 'splunk.spl2.languageServerVersion';
+
+export const stateKeyLatestLspVersion = 'splunk.spl2.latestLspVersion';
+export const stateKeyLastLspCheck = 'splunk.spl2.lastLspCheck';
+
+// Minimum version of Java needed for SPL2 Language Server
+const minimumMajorJavaVersion = 17;
+
+export enum TermsAcceptanceStatus {
+    DeclinedForever = 'declined (forever)',
+    DeclinedOnce = 'declined (once)',
```

Comment:
What does this status mean?

Reply:
This is for when the user hits "Cancel" during install so that the next time they are asked again whether they accept the terms as opposed to "Decline and stop asking me if I want to accept in the future". Essentially it's neither declined nor accepted.

Reply:
maybe `'undecided'` is more clear?

Reply:
Yeah, after your explanation, it makes sense.